docs/src/pages/docs/functions/logic-functions.md

```diff
@@ -308,6 +442,56 @@ function. Either the original or modified value will be returned depending on
 the result of the predicate. Check out [`unless`](#unless) for a negated version
 of this function.
 
+```javascript
+import when from 'crocks/logic/when'
+
+import constant from 'crocks/combinator/constant'
+import flip from 'crocks/combinator/flip'
+import composeB from 'crocks/combinator/composeB'
+import compose from 'crocks/core/compose'
+
+// prod :: Number -> Number -> Number
+const prod = a => b => b * a
+
+// doubleWhen :: (a -> boolean) -> Number -> Number
+const doubleWhen = flip(when, prod(2))
+
+doubleWhen(constant(true), 21)
+//=> 42
+
+doubleWhen(constant(false), 21)
+//=> 21
+
+// gt :: Number -> Number -> Boolean
+const gt = a => b => b > a
+
+// subtract :: Number -> Number
+const subtract = a => b => b - a
+
+// protectedSubtract :: Number -> Number -> Number
+const protectedSubtract = composeB(
+  when(gt(0)), 
+  subtract
+)
+
+// smallExplosion :: Number -> Number
+const smallExplosion = protectedSubtract(15)
+
+// largeExplosion :: Number -> Number
+const largeExplosion = compose(
+  smallExplosion,
+  smallExplosion,
+  smallExplosion,
+)
+
+let health = 30
```

Comment:
I am a little nervous about `let`, is there a way to not use imperative patterns in this example

Reply:
Also need siggys or outputs for every expression

Reply:
yeah there is, will update the pattern

Reply:
I am going to put something basic in here, but if we want to have a bit of fun with it, i could replace it to use a `State` but not sure the extra complexity is valuable in the example. Don't want to add too many layers to it